flang/lib/Lower/IO.cpp

```diff
@@ -196,6 +197,183 @@ static void makeNextConditionalOn(Fortran::lower::FirOpBuilder &builder,
   builder.setInsertionPointToStart(&ifOp.thenRegion().front());
 }
 
+/// Retrieve or generate a runtime description of NAMELIST group `symbol`.
+/// The form of the description is defined in runtime header file namelist.h:
+///
+///     class NamelistGroup {
+///     public:
+///       struct Item {
+///         const char *name; // NUL-terminated lower-case
+///         const Descriptor &descriptor;
+///       };
+///       const char *groupName; // NUL-terminated lower-case
+///       std::size_t items;
+///       const Item *item; // in original declaration order
+///     };
```

Comment:
While I personally find this to be a well-written description, I'm also wondering about including a snippet of code like this because it just isn't something the F18 project really does (or does much). I suppose the argument against it is that there is no way for someone changing `namelist.h` to know that this comment is here.

Reply:
That's a valid point.  I'll change this to just reference the definition, rather than copying it.